### PR TITLE
Added goals and updated spec links to the README.md.

### PR DESCRIPTION
**Pull request by amshuman-kr**
**What this PR does / why we need it**:
Added the scope and the rationale behind behind HVPA in the README.md.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -210,4 +233,14 @@ request
...
    1   3    10     #Replicas
-```
\ No newline at end of file
+```
+
+### Custom Resource Definition
+
+#### Types
+
+See [here](https://github.com/gardener/hvpa-controller/blob/a336e86111bbd5ec87f5a749003ba39b4735261e/api/v1alpha1/hvpa_types.go#L28-L305).
```
Comment (ggaurav10) [inline README.md]: s/a336e86111bbd5ec87f5a749003ba39b4735261e/master/

File: README.md [hunk 2]
```diff
@@ -210,4 +233,14 @@ request
...
+
+#### Example
+
+See [here](https://github.com/gardener/hvpa-controller/blob/a698b08bc371525f666bb2f4766ccb9a403a0d4b/config/samples/autoscaling_v1alpha1_hvpa.yaml).
```
Comment (ggaurav10) [inline README.md]: s/a698b08bc371525f666bb2f4766ccb9a403a0d4b/master

**Review by ggaurav10 — CHANGES_REQUESTED** [review]
Thanks @amshuman-kr for the PR.
Can you please also update/remove the link to the spec [here](https://github.com/gardener/hvpa-controller/blob/master/README.md#spec)

File: README.md (hunk 1 above)
Comment (amshuman-kr) [reply README.md]: Thought `master` would be moving target. Hence, the permalink. WDYT?

File: README.md (hunk 2 above)
Comment (amshuman-kr) [reply README.md]: Thought `master` would be moving target. Hence, the permalink. WDYT?

**Comment by amshuman-kr** [comment]
> Can you please also update/remove the link to the spec [here](https://github.com/gardener/hvpa-controller/blob/master/README.md#spec)

I have fixed the link. Can you PTAL?

**Review by ggaurav10 — APPROVED** [review]
LGTM

**Pull request merged**